src/main/cpp/src/json_parser.hpp

```diff
@@ -0,0 +1,1833 @@
+              *copy_dest++ = '\\';
+            }
+            bytes_diff_for_escape_writing++;
+          }
+          if (!try_match_char(to_match_str_pos, to_match_str_end, c)) { return false; }
+          string_token_utf8_bytes++;
+          str_pos++;
+          return true;
+        case '/':
+          // for both unescaped/escaped writes a single char /
+          if (nullptr != copy_dest) { *copy_dest++ = c; }
+          if (!try_match_char(to_match_str_pos, to_match_str_end, c)) { return false; }
+          string_token_utf8_bytes++;
+          str_pos++;
+          return true;
+        case 'b':
+          if (nullptr != copy_dest && write_style::unescaped == w_style) { *copy_dest++ = '\b'; }
+          if (write_style::escaped == w_style) {
+            if (copy_dest != nullptr) {
+              *copy_dest++ = '\\';
+              *copy_dest++ = 'b';
+            }
+            bytes_diff_for_escape_writing++;
+          }
+          if (!try_match_char(to_match_str_pos, to_match_str_end, '\b')) { return false; }
+          string_token_utf8_bytes++;
+          str_pos++;
+          return true;
+        case 'f':
+          if (nullptr != copy_dest && write_style::unescaped == w_style) { *copy_dest++ = '\f'; }
+          if (write_style::escaped == w_style) {
+            if (copy_dest != nullptr) {
+              *copy_dest++ = '\\';
+              *copy_dest++ = 'f';
+            }
+            bytes_diff_for_escape_writing++;
+          }
+          if (!try_match_char(to_match_str_pos, to_match_str_end, '\f')) { return false; }
+          string_token_utf8_bytes++;
+          str_pos++;
+          return true;
+        case 'n':
+          if (nullptr != copy_dest && write_style::unescaped == w_style) { *copy_dest++ = '\n'; }
+          if (write_style::escaped == w_style) {
+            if (copy_dest != nullptr) {
+              *copy_dest++ = '\\';
+              *copy_dest++ = 'n';
+            }
+            bytes_diff_for_escape_writing++;
+          }
+          if (!try_match_char(to_match_str_pos, to_match_str_end, '\n')) { return false; }
+          string_token_utf8_bytes++;
+          str_pos++;
+          return true;
+        case 'r':
+          if (nullptr != copy_dest && write_style::unescaped == w_style) { *copy_dest++ = '\r'; }
+          if (copy_dest != nullptr && write_style::escaped == w_style) {
+            if (copy_dest != nullptr) {
+              *copy_dest++ = '\\';
+              *copy_dest++ = 'r';
+            }
+            bytes_diff_for_escape_writing++;
+          }
+          if (!try_match_char(to_match_str_pos, to_match_str_end, '\r')) { return false; }
+          string_token_utf8_bytes++;
+          str_pos++;
+          return true;
+        case 't':
+          if (nullptr != copy_dest && write_style::unescaped == w_style) { *copy_dest++ = '\t'; }
+          if (copy_dest != nullptr && write_style::escaped == w_style) {
+            if (copy_dest != nullptr) {
```

Comment:
Check again: copy_dest != nullptr, it is redudant.
Go through all `copy_dest` and `w_style` related paths.
And add cases.